File: Golang/UniqueBinarySearchTrees/UniqueBinarySearchTrees_test.go
package UniqueBinarySearchTrees

import "testing"

func TestNumTrees(t *testing.T) {

	if numTrees(0) == 1 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if numTrees(1) == 1 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if numTrees(2) == 2 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if numTrees(3) == 5 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if numTrees(4) == 14 {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/UniqueBinarySearchTreesTest.py
from unittest import TestCase

from UniqueBinarySearchTrees import UniqueBinarySearchTrees


class TestUniqueBinarySearchTrees(TestCase):
    def test_numTrees(self):
        u = UniqueBinarySearchTrees()

        self.assertTrue(u.numTrees(0) == 1)

        self.assertTrue(u.numTrees(1) == 1)

        self.assertTrue(u.numTrees(2) == 2)

        self.assertTrue(u.numTrees(3) == 5)

        self.assertTrue(u.numTrees(4) == 14)
